### DISCUSSION TITLE
Finding a pydantic line of code replacement (pydantic.json.ENCODERS_BY_TYPE)

### DISCUSSION THREAD

**Discussion by vivek378521** [pydantic/pydantic#7090] [Pydantic V2]
We have the below loc

`# Automatically convert ObjectId to str when sending back results`
`pydantic.json.ENCODERS_BY_TYPE[ObjectId] = str`

It breaks when we update package to fastapi=0.100 and pydantic=2.*

for pydantic v2 what is the replacement for this?

**Comment by bahadiraraz** [comment]
I tried many methods but none of them worked. Can someone write how we will do it? I can't migrate my project to FastAPI and Pydantic just because of this line.

**Comment by samuelcolvin** [comment]
The general solution is to use `field_serialiser`, or implement `__get_pydantic_core_schema__`on your custom types.

If that doesn't work for you, please explain more about your problem.

**Reply by vivek378521** [reply]
It is a global declaration, I cannot find any example of replacing this particular loc. 

Whenever I run my code after upgrading to v2, I get this error

`File "/app/app/__init__.py", line 13, in <module>
    from app.settings import Settings
  File "/app/app/settings.py", line 3, in <module>
    from app.utils import urljoin
  File "/app/app/utils.py", line 14, in <module>
    pydantic.json.ENCODERS_BY_TYPE[ObjectId] = str
  File "/usr/local/lib/python3.10/site-packages/pydantic/__init__.py", line 363, in __getattr__
    return _getattr_migration(attr_name)
  File "/usr/local/lib/python3.10/site-packages/pydantic/_migration.py", line 306, in wrapper
    raise AttributeError(f'module {module!r} has no attribute {name!r}')
AttributeError: module 'pydantic' has no attribute 'json'. Did you mean: 'Json'?`

**Comment by bahadiraraz** [comment]
```python
if isinstance(user.get("_id"), ObjectId):
    user["_id"] = str(user["_id"])
```
My problem is solved with this line 🤙

**Reply by vivek378521** [reply]
the line of code that we used above is global in nature, it automatically converts all object ids to str, now we have to search for something new. I was wondering if we can just replace the line with something that is available in pydantic v2 it would be easier.

**Reply by bahadiraraz** [reply]
maybe u can use
 `from pydantic.v1.json import ENCODERS_BY_TYPE`

then continue with your code ENCODERS_BY_TYPE[ObjectId] = str
